src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -134,34 +111,16 @@ export class TransactionsHistoryMapper {
     return transactions[0];
   }
 
-  private getDayFromTransactionDate(
-    transaction: TransactionDomain,
-    timezoneOffset: number,
-  ): Date {
-    const timestamp = this.getTransactionTimestamp(transaction);
-    return this.getDayStartForDate(timestamp, timezoneOffset);
-  }
-
   private groupByDay(
-    transactions: TransactionDomain[],
+    transactions: TransactionItem[],
     timezoneOffset: number,
-  ): TransactionDomainGroup[] {
-    return Object.entries(
-      groupBy(transactions, (transaction) => {
-        return this.getDayFromTransactionDate(
-          transaction,
-          timezoneOffset,
-        ).getTime();
-      }),
-    ).map(([, transactions]): TransactionDomainGroup => {
-      // The groups respect the timezone offset – this was done for grouping only.
-      // The actual value of the group should be in the UTC timezone instead
-      // A group should always have at least one transaction.
-      return {
-        timestamp: this.getTransactionTimestamp(transactions[0]).getTime(),
-        transactions: transactions,
-      };
+  ): TransactionItem[][] {
+    const grouped = groupBy(transactions, ({ transaction }) => {
+      // timestamp will always be defined for historical transactions
+      const date = new Date(transaction.timestamp ?? 0);
```

Comment:
Have you considered here `CreationTransaction`? Those won't include a `timestamp` but a `created` field holding its timestamp I think.

Reply:
As the transactions are first mapped, [`created` is mapped as `timestamp`](https://github.com/safe-global/safe-client-gateway/blob/refactor-history-mapping/src/routes/transactions/mappers/creation-transaction/creation-transaction.mapper.ts#L46) here.